src/Analyser/NodeScopeResolver.php

```diff
@@ -4098,6 +4108,53 @@ public function getPhpDocs(Scope $scope, Node\FunctionLike|Node\Stmt\Property $n
 		return [$templateTypeMap, $phpDocParameterTypes, $phpDocReturnType, $phpDocThrowType, $deprecatedDescription, $isDeprecated, $isInternal, $isFinal, $isPure, $acceptsNamedArguments, $isReadOnly, $docComment, $asserts];
 	}
 
+	/**
+	 * @return array<string, Type>
+	 */
+	private function getParameterOutTypes(FunctionReflection|MethodReflection|null $reflection, Scope $scope): array
+	{
+		if ($reflection === null) {
+			return [];
+		}
+
+		$class = null;
+		$trait = null;
+		if ($reflection instanceof MethodReflection) {
+			$class = $reflection->getDeclaringClass()->getName();
+			$file = $reflection->getDeclaringClass()->getFileName();
+		} else {
+			$file = $reflection->getFileName();
+		}
+
+		$resolvedPhpDoc = null;
+		$functionName = $reflection->getName();
+		$docComment = $reflection->getDocComment();
+
+		if ($docComment !== null) {
+			$resolvedPhpDoc = $this->fileTypeMapper->getResolvedPhpDoc(
+				$file,
+				$class,
+				$trait,
+				$functionName,
+				$docComment,
+			);
+		}
+
+		if ($resolvedPhpDoc === null) {
+			return [];
+		}
+
+		$phpDocParameterOutTypes = [];
+		foreach ($resolvedPhpDoc->getParamOutTags() as $paramName => $paramOutTag) {
+			$paramOutType = $paramOutTag->getType();
+			if ($scope->isInClass()) {
+				$paramOutType = $this->transformStaticType($scope->getClassReflection(), $paramOutType);
+			}
```

Comment:
I think this one here is not correct (copy paste leftover). need to find a proper testcase first..